test/util/bitcoin-util-test.py

```diff
@@ -77,32 +77,36 @@ def bctest(testDir, testObj, buildenv):
     are not as expected. Error is caught by bctester() and reported.
     """
     # Get the exec names and arguments
-    execprog = buildenv["BUILDDIR"] + "/src/" + testObj['exec'] + buildenv["EXEEXT"]
+    execprog = os.path.join(buildenv["BUILDDIR"], "src", testObj["exec"]) + buildenv["EXEEXT"]
     execargs = testObj['args']
     execrun = [execprog] + execargs
 
     # Read the input data (if there is any)
     stdinCfg = None
     inputData = None
     if "input" in testObj:
-        filename = testDir + "/" + testObj['input']
+        filename = os.path.join(testDir, testObj["input"])
         inputData = open(filename).read()
         stdinCfg = subprocess.PIPE
 
     # Read the expected output data (if there is any)
     outputFn = None
     outputData = None
+    outputType = None
     if "output_cmp" in testObj:
         outputFn = testObj['output_cmp']
         outputType = os.path.splitext(outputFn)[1][1:]  # output type from file extension (determines how to compare)
         try:
-            outputData = open(testDir + "/" + outputFn).read()
+            outputData = open(os.path.join(testDir, outputFn)).read()
         except:
             logging.error("Output file " + outputFn + " can not be opened")
             raise
         if not outputData:
             logging.error("Output data missing for " + outputFn)
             raise Exception
+        if not outputType:
+            logging.error("Output type missing for " + outputFn)
```

Comment:
suggestion: change the error text slightly here to clarify what the problem is:

```python
logging.error("Output file %s does not have a file extension" % outputFn)
```

Reply:
@dongsam Could you address @jnewbery's nit here? Looks like this is pretty much ready for merge.

Reply:
@fanquake I just change and rebase the error text, sorry for late response